falcon/testing/client.py

```diff
@@ -2133,8 +2192,163 @@ async def __aexit__(self, exc_type, exc, tb):
         await self._task_req
 
 
+def _prepare_data_fields(data, boundary=None, urlenc=False):
+    """Prepare data  fields for request body.
+
+    Args:
+        data: dict or list of tuples with json data from the request
+
+    Returns: list of 2-tuples (field-name(str), value(bytes))
+
+    """
+    urlresult = []
+    body_part = b''
+    if isinstance(data, (str, bytes)) or hasattr(data, 'read'):
+        fields = list(json_module.loads(data).items())
```

Comment:
so we are assuming that if provided as string it's a json.

this would mean that we cannot replace deprecate body with data.

let's see @vytas7 opinion

Reply:
ok, I hadn't really understood what you guys meant by replacing the body... but ok, that should be an easy fix. I'll try to implement it, then you can use whatever works best.